### PR TITLE
instrument interaction events

### PR DESCRIPTION
**Pull request by jackson-dean**

### CODE REVIEW

File: src/app/buy-deso/buy-deso-megaswap/buy-deso-megaswap.component.html [hunk 1]
```diff
@@ -11,6 +11,7 @@
   </ng-template>
   <div class="d-flex justify-content-center">
     <button
+      data-control-name="buy-deso-finish-flow-button"
```
Comment (lazynina) [inline src/app/buy-deso/buy-deso-megaswap/buy-deso-megaswap.component.html:14]: I think we've got a public key at this point - should we log the public key of the user who finished the flow?

File: src/app/buy-or-send-deso/buy-or-send-deso.component.html [hunk 2]
```diff
@@ -31,6 +31,7 @@
           <div class="d-flex justify-content-around">
             <div class="col-10 d-flex align-items-center flex-column">
               <button
+                data-control-name="buy-or-send-buy-now-button"
```
Comment (lazynina) [inline src/app/buy-or-send-deso/buy-or-send-deso.component.html:34]: I won't comment on each of these, but I think everywhere we have a public key, we should log it in the interaction event.

**Review by lazynina — COMMENTED** [review]
Look good other than logging public keys where possible.

File: src/app/buy-deso/buy-deso-megaswap/buy-deso-megaswap.component.html (hunk 1 above)
Comment (jackson-dean) [reply src/app/buy-deso/buy-deso-megaswap/buy-deso-megaswap.component.html:14]: yep i added a thing to just always pull it from the query params. if it's not there yet it'll just be empty which seems fine. The whole stream of interactions should be back associated once diamond identifies the user after login.

File: src/app/buy-or-send-deso/buy-or-send-deso.component.html (hunk 2 above)
Comment (jackson-dean) [reply src/app/buy-or-send-deso/buy-or-send-deso.component.html:34]: I think the global query param parser thing should take care of this.

File: src/app/backend-api.service.ts [hunk 3]
```diff
@@ -288,6 +289,13 @@ export class BackendAPIService {
 
   jwtPost(path: string, publicKey: string, body: any): Observable<any> {
     const publicUserInfo = this.accountService.getEncryptedUsers()[publicKey];
+    // NOTE: there are some cases were derived user's were not being sent phone number
```
Comment (lazynina) [inline src/app/backend-api.service.ts:292]:
```suggestion
    // NOTE: there are some cases where derived user's were not being sent phone number
```

**Review by lazynina — APPROVED** [review]

**Pull request merged**